src/components/pages/maintenance/ruleDetails/ruleDetails.js

```diff
@@ -1,12 +1,248 @@
 // Copyright (c) Microsoft. All rights reserved.
 
-import React from 'react';
+import React, { Component } from 'react';
+import { Subject } from 'rxjs';
 
-import { PageContent, ContextMenu } from 'components/shared';
+import Config from 'app.config';
+import { RulesGrid } from 'components/pages/rules/rulesGrid';
+import { PageContent, ContextMenu, RefreshBar } from 'components/shared';
+import { joinClasses } from 'utilities';
+import { DevicesGrid } from 'components/pages/devices/devicesGrid';
+import { TelemetryChart, transformTelemetryResponse } from 'components/pages/dashboard/panels/telemetry';
+import { TelemetryService } from 'services';
+import { LastTriggerRenderer, SeverityRenderer } from 'components/shared/cellRenderers';
+import { AlarmOccurrencesGrid } from 'components/pages/maintenance/grids';
 
-export const RuleDetails = ({ match: { params: { id } } }) => [
-  <ContextMenu key="context-menu"></ContextMenu>,
-  <PageContent className="maintenance-container" key="page-content">
-    <div className="header">{id}</div>
-  </PageContent>
+import './ruleDetails.css';
+
+const renderCount = (value) => typeof value === 'undefined' ? '--' : value;
+
+// TODO: Move colors to the redux store
+const chartColors = [
+  '#01B8AA',
+  '#F2C80F',
+  '#E81123',
+  '#3599B8',
+  '#33669A',
+  '#26FFDE',
+  '#E0E7EE',
+  '#FDA954',
+  '#FD625E',
+  '#FF4EC2',
+  '#FFEE91'
 ];
+
+const idDelimiter = ' ';
+
+export class RuleDetails extends Component {
```

Comment:
Is this different than the rules details flyout?

Reply:
Yes, this is the rules view in the maintenance page.